tests/miral/wayland_extensions.cpp

```diff
@@ -180,12 +180,16 @@ struct ClientDecorationCreator
         wl_display_roundtrip(display);
         wl_display_roundtrip(display);
 
+        ASSERT_THAT(compositor, NotNull());
+        ASSERT_THAT(decoration_manager, NotNull());
+        ASSERT_THAT(shell, NotNull());
```

Comment:
My thought was that other tests that don't use this should be allowed to run even if something is seriously broken with what wayland interfaces are exposed.

Reply:
If these fail, then the problem is not with the code being tested here and should not be reported as such.

Reply:
It's obvious what the failures mean though. There will be a big line of identical failures, and they lead here upon any investigation. It wont waste a significant amount of development time, unlike aborting the program which makes the tests increasingly heard to work with.

Imagine you're porting to a new platform or something, a lot of tests are failing but half of them are aborting the program instead of reporting failures. You'd have to do a bunch of --gtest_filter trial-and-error just to see the full list of what's failing.